### PR TITLE
Refactor TransportShardBulkAction#executeBulkItemRequest

### PR DESCRIPTION
**Pull request by dakrone**
This refactors the `TransportShardBulkAction#executeBulkItemRequest` method into
multiple methods so that it can be more easily read. It does not change the
behavior in any way.

(This is a precursor to unit testing `TransportShardBulkAction`)

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java [hunk 1]
```diff
@@ -237,29 +278,16 @@ private static boolean isConflictException(final Exception e) {
         return ExceptionsHelper.unwrapCause(e) instanceof VersionConflictEngineException;
     }
 
-    private static class UpdateResultHolder {
```
Comment (areek) [inline core/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java]: +1

File: core/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java [hunk 2]
```diff
@@ -123,6 +125,102 @@ protected boolean resolveIndex() {
...
+        } else {
+            // update the request with the version so it will go to the replicas
+            deleteRequest.versionType(deleteRequest.versionType().versionTypeForReplicationAndRecovery());
+            deleteRequest.version(deleteResult.getVersion());
+            deleteRequest.setSeqNo(deleteResult.getSeqNo());
+            assert deleteRequest.versionType().validateVersionForWrites(deleteRequest.version());
+            return new BulkItemResultHolder(new DeleteResponse(primary.shardId(), deleteRequest.type(), deleteRequest.id(),
+                                                               deleteResult.getSeqNo(), deleteResult.getVersion(),
+                                                               deleteResult.isFound()),
+                    deleteResult, bulkItemRequest);
```
Comment (areek) [inline core/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java]: nit: indentation is off

File: core/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java [hunk 3]
```diff
@@ -123,6 +125,102 @@ protected boolean resolveIndex() {
...
+            // update the version on request so it will happen on the replicas
+            final long version = indexResult.getVersion();
+            indexRequest.version(version);
+            indexRequest.versionType(indexRequest.versionType().versionTypeForReplicationAndRecovery());
+            indexRequest.setSeqNo(indexResult.getSeqNo());
+            assert indexRequest.versionType().validateVersionForWrites(indexRequest.version());
+            return new BulkItemResultHolder(new IndexResponse(primary.shardId(), indexRequest.type(), indexRequest.id(),
+                                                              indexResult.getSeqNo(), indexResult.getVersion(),
+                                                              indexResult.isCreated()),
+                    indexResult, bulkItemRequest);
```
Comment (areek) [inline core/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java]: nit: indentation is off

**Review by areek — COMMENTED** [review]
This looks good, IMO we should wait for https://github.com/elastic/elasticsearch/pull/23038 to be merged first. I wonder how this will look after merging #23038, given we don't have update the write requests anymore

**Comment by bleskes** [comment]
>I wonder how this will look after merging #23038, given we don't have update the write requests anymore 

+1

**Comment by dakrone** [comment]
> I wonder how this will look after merging #23038, given we don't have update the write requests anymore

I'll probably redo this entirely and open a new PR

**Comment by dakrone** [comment]
Closing this, I'll re-open a different PR

**Pull request closed without merging**